src/main/java/picard/sam/BamErrorMetric/CollectBamErrorMetrics.java

```diff
@@ -0,0 +1,470 @@
+                "<p>" +
+                "The BASE_QUALITY stratifier will place bases in bins according to their declared base quality. " +
+                "The READ_ORDINALITY stratifier will place bases in one of two bins depending on whether their read " +
+                "is 'first' or 'second'. One could generate a composite stratifier BASE_QUALITY:READ_ORDINALITY which will " +
+                "do both stratifications as the same time. \n" +
+                "<p>" +
+                "The resulting metric file will be named according to a provided prefix and a suffix which is generated " +
+                " automatically according to the error metric. " +
+                "The tool cal collect multiple metrics in a single pass and there should be hardly any " +
+                "performance loss when specifying multiple metrics at the same time; the default includes a " +
+                "large collection of metrics. \n" +
+                "<p>" +
+                "To estimate the error rate the tool assumes that all differences from the reference are errors. For this to be " +
+                "a reasonable assumption the tool needs to know the sites at which the sample is actually polymorphic and a " +
+                "confidence interval where the user is relatively certain that the polymorphic sites are known and accurate. " +
+                "These two inputs are provided as a VCF and INTERVALS. The program will only process sites that are in the " +
+                "intersection of the interval lists in the INTERVALS argument as long as they are not polymorphic in the " +
+                "VCF.\n\n",
+        oneLineSummary = "Program to collect error metrics on bases stratified in various ways.",
+        programGroup = DiagnosticsAndQCProgramGroup.class
+)
+public class CollectBamErrorMetrics extends CommandLineProgram {
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "Input SAM or BAM file.")
+    public File INPUT;
+
+    @Argument(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "Base name for output files. Actual file names will be " +
+            "generated from the basename and suffixes from the ERROR and STRATIFIER by adding a '.' and then " +
+            "error_by_stratifier[_and_stratifier]* where 'error' is ERROR's extension, and 'stratifier' is STRATIFIER's suffix. " +
+            "For example, an ERROR_METRIC of ERROR:BASE_QUALITY:GC_CONTENT will produce an extension '.error_by_base_quality_and_gc'. " +
+            "The suffixes can be found in the documentation for ERROR_VALUE and SUFFIX_VALUE." )
+    public File OUTPUT;
+
+    @Argument(doc = "Errors to collect in the form of \"ERROR(:STRATIFIER)*\". " +
+            "To see the values available for ERROR and STRATIFIER look at the documentation for the arguments " +
+            "ERROR_VALUE and STRATIFIER_VALUE.")
+    public List<String> ERROR_METRICS = new ArrayList<>(CollectionUtil.makeList(
```

Comment:
Is there a reason why `Arrays.asList()` isn't used here? It doesn't look like this is modified after its creation.

Reply:
It is modified by the magic of the commandline program parsing system.

Reply:
OK, thought it might be that. In that case, is the `new ArrayList<>()` needed too? It looks like `CollectionUtil.makeList()` returns a modifiable list already.

Reply:
true. done.